pkg/controller/storagecluster/initialization_reconciler_test.go

```diff
@@ -392,39 +408,57 @@ func TestEnsureExternalStorageClusterResources(t *testing.T) {
 			},
 		},
 	}
-	jsonBlob, err := json.Marshal(ExternalResources)
+	externalSecret, err := createExternalCephClusterSecret()
 	if err != nil {
-		t.Fatalf("fatal err %+v", err)
+		t.Fatalf("failed to create external secret: %v", err)
+	}
+	reconciler := createFakeInitializationStorageClusterReconciler(t, &nbv1.NooBaa{})
+	runtimeObjs := []runtime.Object{cr, externalSecret}
+	for _, eachRTObj := range runtimeObjs {
+		if err = reconciler.client.Create(nil, eachRTObj); err != nil {
+			break
```

Comment:
Why are you doing a break here? Shouldn't you do a `t.Fatalf()`?

Reply:
Previously, the function was returning a pair of `reconciler` and  `error` and the break was remains of the previous code. Thanks to catch it. (=
Changed it.